src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1091,6 +1095,34 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 					AiNodeResetParameter( node, g_lightGroupArnoldString );
 					AiNodeResetParameter( node, g_useLightGroupArnoldString );
 				}
+
+				if( m_linkedShadows )
+				{
+					// NOTE : We introduce even more conversions and lookups at this
```

Comment:
Would be worth putting this logic into a function and reusing it, rather than duplicating the light link code.